streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -40,9 +43,27 @@
  * A join is symmetric in the sense, that a join specification on the first stream returns the same result record as
  * a join specification on the second stream with flipped before and after values.
  * <p>
- * Both values (before and after) must not result in an "inverse" window,
- * i.e., lower-interval-bound must not be larger than upper-interval.bound.
+ * Both values (before and after) must not result in an "inverse" window, i.e., upper-interval-bound cannot be smaller
+ * than lower-interval bound.
+ * <p>
+ * {@link JoinWindows} are sliding windows, thus, they are aligned to the actual record timestamps.
+ * This implies, that each input record defines its own window with start and end time being relative to the record's
+ * timestamp.
+ * <p>
+ * For time semantics, compare {@link org.apache.kafka.streams.processor.TimestampExtractor TimestampExtractor}.
```

Comment:
This is not clear to me: "compare TimestampExtractor"? Or rather "see TimestampExtractor"?